ts/src/base/Exchange.ts

```diff
@@ -2722,6 +2722,14 @@ export default class Exchange {
         return res === 0;
     }
 
+    safeNumberOmitZero (obj: object, key: IndexType, defaultValue: Num = undefined): Num {
+        const value = this.safeNumber (obj, key, defaultValue);
+        if (value === 0 || (typeof value === 'string' && Precise.stringEq (value, '0'))) {
```

Comment:
i think this approach is optimal, compared to parsing number as a string at first, then parsing back to number. to avoid that, we can have just type check to avoid extra calcs. not?